src/App.jsx

```diff
@@ -14,33 +15,81 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const copyGoods = [...goodsFromServer];
+  const [sortType, setSort] = useState('');
+  const [isReversed, setReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  copyGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case 'Sort by length':
+        return good1.length - good2.length;
+      case 'Sort alphabetically':
+        return good1.localeCompare(good2);
+      default:
+        return 0;
+    }
+  });
+  if (isReversed) {
+    copyGoods.reverse();
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const getButtonClass = buttonType => {
+    return sortType === buttonType ? '' : 'is-light';
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const reset = () => {
+    setReversed(false);
+    setSort('');
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSort('Sort alphabetically')}
+          type="button"
+          className={`button is-info ${getButtonClass('Sort alphabetically')}`}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSort('Sort by length')}
+          type="button"
+          className={`button is-success ${getButtonClass('Sort by length')}`}
+        >
+          Sort by length
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          onClick={() => setReversed(!isReversed)}
+          type="button"
+          className={`button is-warning ${isReversed ? '' : 'is-light'}`}
+        >
+          Reverse
+        </button>
+
+        {(sortType || isReversed) && (
+          <button
+            onClick={() => {
+              reset();
+            }}
```

Comment:
There is no need to create a function that simply calls another function. This can be done directly in the onClick property by passing a reference to the reset function

```suggestion
          <button
            onClick={reset}
```